lib/ansible/cli/galaxy.py

```diff
@@ -76,20 +99,19 @@ def _display_role(gr):
 
 def _display_collection(collection, cwidth=10, vwidth=7, min_cwidth=10, min_vwidth=7):
     display.display('{fqcn:{cwidth}} {version:{vwidth}}'.format(
-        fqcn=to_text(collection),
-        version=collection.latest_version,
+        fqcn=to_text(collection.fqcn),
+        version=collection.ver,
```

Comment:
@felixfontein could you check if making this
```suggestion
        version=collection.ver or '*',
```
would fix `list` in your env? It'd be great to know how to reproduce/debug this.

Reply:
Yes, that actually fixes it. The collection printed there is `ansible.netcommon`, which is the latest git checkout from https://github.com/ansible-collections/ansible.netcommon

Maybe it's because it has an explicit `version: null` in its galaxy.yml: https://github.com/ansible-collections/ansible.netcommon/blob/main/galaxy.yml#L13. But when I comment `version: null` out, `ansible-galaxy collection list` simply will not work (both in `devel` and this branch):
```
ERROR! The collection galaxy.yml at '/path/to/ansible_collections/ansible/netcommon/galaxy.yml' is missing the following mandatory keys: version
```

Reply:
> Maybe it's because it has an explicit `version: null` in its galaxy.yml

Oh, that would make sense. `dict.get('version', '*')` wouldn't fall back to `*` in case if the key exists but is `None`. Thanks for the pointer!

Reply:
That should better be written as `dict.get('version') or '*'`.

Reply:
Yeah, probably. I'm just trying to figure out on what level this should be fixed and if there are other cases where this problem would happen if I just patch here.

Reply:
I attempted to fix this in https://github.com/ansible/ansible/pull/70185, but that ended up being way too invasive. Since `_display_collection()` is only used when listing collection information, I think it's fine to do `collection.get(ver) or '*'` here.

Reply:
@samdoran it's already been fixed elsewhere to make sure that missing or falsy version is turned into `*` when the metadata is read.